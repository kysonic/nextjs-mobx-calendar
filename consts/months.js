export const DAYS = ['Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday', 'Sunday'];
export const MONTHS = ['January', 'February', 'March', 'April', 'May', 'June', 'July', 'August', 'September', 'October', 'November', 'December'];
export const DAYS_IN_MONTH = 31;
export const DAYS_IN_WEEK = 7;
