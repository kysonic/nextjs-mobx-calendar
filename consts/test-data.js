export const APP_INITIAL_DATA = {
    navigationStore: {
        items: [
            {
                "id": "home",
                "title": "Home",
                "href": "/"
            }
        ]
    },
    dateStore: {
        ISODate: '2019-01-01T00:00:00.000Z'
    }
};

export const CALENDAR_INITIAL_DATA = {
    "2019-04-10T00:00:00.000Z": [
        {
            "id": "abcde1",
            "title": "Action 1",
            "description": "Description of action 1"
        },
        {
            "id": "abcde2",
            "title": "Action 2",
            "description": "Description of action 2"
        }
    ],
    "2019-03-11T00:00:00.000Z": [
        {
            "id": "abcde2",
            "title": "Action 2",
            "description": "Description of action 2"
        },
        {
            "id": "abcde3",
            "title": "Action 4",
            "description": "Description of action 4"
        }
    ]
}
