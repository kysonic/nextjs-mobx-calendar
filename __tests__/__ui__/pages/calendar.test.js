import CalendarPage from '../../../pages/calendar';
import renderer from 'react-test-renderer';
import {Provider} from 'mobx-react/index';
import AppStore from '../../../stores/app';
import {APP_INITIAL_DATA} from '../../../consts/test-data';
import axios from 'axios/index';
import ReactDOM from "react-dom";
import {mockCreatePortal} from '../../__mocks__/react-dom.mock';

const CALENDAR_DATA = {
    "2019-04-10T00:00:00.000Z": [
        {
            "id": "abcde1",
            "title": "Action 1",
            "description": "Description of action 1"
        },
        {
            "id": "abcde2",
            "title": "Action 2",
            "description": "Description of action 2"
        }
    ]
};

// Mocks
jest.mock('axios');
axios.get.mockResolvedValue({data: CALENDAR_DATA});
jest.mock('react-dom');
ReactDOM.createPortal = jest.fn(mockCreatePortal);

describe('Calendar page tests', () => {
    test('Renders without crashes', () => {
        const appStore = new AppStore(APP_INITIAL_DATA);
        const tree = renderer.create(
            <Provider appStore={appStore}>
                <CalendarPage initialState={[{items: CALENDAR_DATA}]}/>
            </Provider>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});

