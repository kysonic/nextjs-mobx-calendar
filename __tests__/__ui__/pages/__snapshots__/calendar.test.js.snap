// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calendar page tests Renders without crashes 1`] = `
<div
  className="jsx-3569970921"
>
  <nav
    className="jsx-3989381672"
  >
    <ul
      className="jsx-3989381672"
    >
      <li
        className="jsx-3989381672"
      >
        <span
          onClick={[Function]}
        >
          Home
        </span>
      </li>
    </ul>
    <style>
      
                     ul li:hover .sub-items {
                        display: block;
                     }
                     ul li a, ul li a:visited, ul li a:active {
                        color: #666;
                     }
                     ul li:hover a {
                        color: #444;
                     }
                
    </style>
  </nav>
  <div
    className="jsx-3569970921 content"
  >
    <div
      className="calendar"
    >
      <div
        className="jsx-3220420617 month-switcher"
        data-testid="month-switcher"
      >
        <div
          className="jsx-3220420617 arrow arrow-left"
          data-testid="arrow-left"
          onClick={[Function]}
          role="button"
          tabIndex="0"
        >
          &lt;
        </div>
        <div
          className="jsx-3220420617 month"
          data-testid="month"
        >
          January
        </div>
        <div
          className="jsx-3220420617 arrow arrow-right"
          data-testid="arrow-right"
          onClick={[Function]}
          role="button"
          tabIndex="0"
        >
          &gt;
        </div>
      </div>
      <table
        className="jsx-3390703719 calendar-table"
        data-testid="table"
      >
        <thead
          className="jsx-3390703719"
          data-testid="table-head"
        >
          <tr
            className="jsx-3390703719"
          >
            <th
              className="jsx-3390703719"
              data-testid="Monday"
            >
              Monday
            </th>
            <th
              className="jsx-3390703719"
              data-testid="Tuesday"
            >
              Tuesday
            </th>
            <th
              className="jsx-3390703719"
              data-testid="Wednesday"
            >
              Wednesday
            </th>
            <th
              className="jsx-3390703719"
              data-testid="Thursday"
            >
              Thursday
            </th>
            <th
              className="jsx-3390703719"
              data-testid="Friday"
            >
              Friday
            </th>
            <th
              className="jsx-3390703719"
              data-testid="Saturday"
            >
              Saturday
            </th>
            <th
              className="jsx-3390703719"
              data-testid="Sunday"
            >
              Sunday
            </th>
          </tr>
        </thead>
        <tbody
          className="jsx-3390703719"
        >
          <tr
            className="jsx-3390703719"
          >
            <td
              className="jsx-3390703719"
              data-testid="table-cell-0-0"
            >
              
            </td>
            <td
              className="jsx-3390703719"
              data-testid="table-cell-0-1"
            >
              <div
                className="jsx-3877038798 calendar-cell is-today"
              >
                <div
                  className="jsx-3877038798 day"
                >
                  1
                </div>
                <div
                  className="jsx-3877038798 plus"
                  data-testid="add-event-2019-01-02T00:00:00.000Z"
                  onClick={[Function]}
                  role="button"
                >
                  +
                </div>
              </div>
            </td>
            <td
              className="jsx-3390703719"
              data-testid="table-cell-0-2"
            >
              <div
                className="jsx-3877038798 calendar-cell "
              >
                <div
                  className="jsx-3877038798 day"
                >
                  2
                </div>
                <div
                  className="jsx-3877038798 plus"
                  data-testid="add-event-2019-01-03T00:00:00.000Z"
                  onClick={[Function]}
                  role="button"
                >
                  +
                </div>
              </div>
            </td>
            <td
              className="jsx-3390703719"
              data-testid="table-cell-0-3"
            >
              <div
                className="jsx-3877038798 calendar-cell "
              >
                <div
                  className="jsx-3877038798 day"
                >
                  3
                </div>
                <div
                  className="jsx-3877038798 plus"
                  data-testid="add-event-2019-01-04T00:00:00.000Z"
                  onClick={[Function]}
                  role="button"
                >
                  +
                </div>
              </div>
            </td>
            <td
              className="jsx-3390703719"
              data-testid="table-cell-0-4"
            >
              <div
                className="jsx-3877038798 calendar-cell "
              >
                <div
                  className="jsx-3877038798 day"
                >
                  4
                </div>
                <div
                  className="jsx-3877038798 plus"
                  data-testid="add-event-2019-01-05T00:00:00.000Z"
                  onClick={[Function]}
                  role="button"
                >
                  +
                </div>
              </div>
            </td>
            <td
              className="jsx-3390703719"
              data-testid="table-cell-0-5"
            >
              <div
                className="jsx-3877038798 calendar-cell "
              >
                <div
                  className="jsx-3877038798 day"
                >
                  5
                </div>
                <div
                  className="jsx-3877038798 plus"
                  data-testid="add-event-2019-01-06T00:00:00.000Z"
                  onClick={[Function]}
                  role="button"
                >
                  +
                </div>
              </div>
            </td>
            <td
              className="jsx-3390703719"
              data-testid="table-cell-0-6"
            >
              <div
                className="jsx-3877038798 calendar-cell "
              >
                <div
                  className="jsx-3877038798 day"
                >
                  6
                </div>
                <div
                  className="jsx-3877038798 plus"
                  data-testid="add-event-2019-01-07T00:00:00.000Z"
                  onClick={[Function]}
                  role="button"
                >
                  +
                </div>
              </div>
            </td>
          </tr>
          <tr
            className="jsx-3390703719"
          >
            <td
              className="jsx-3390703719"
              data-testid="table-cell-1-0"
            >
              <div
                className="jsx-3877038798 calendar-cell "
              >
                <div
                  className="jsx-3877038798 day"
                >
                  7
                </div>
                <div
                  className="jsx-3877038798 plus"
                  data-testid="add-event-2019-01-08T00:00:00.000Z"
                  onClick={[Function]}
                  role="button"
                >
                  +
                </div>
              </div>
            </td>
            <td
              className="jsx-3390703719"
              data-testid="table-cell-1-1"
            >
              <div
                className="jsx-3877038798 calendar-cell "
              >
                <div
                  className="jsx-3877038798 day"
                >
                  8
                </div>
                <div
                  className="jsx-3877038798 plus"
                  data-testid="add-event-2019-01-09T00:00:00.000Z"
                  onClick={[Function]}
                  role="button"
                >
                  +
                </div>
              </div>
            </td>
            <td
              className="jsx-3390703719"
              data-testid="table-cell-1-2"
            >
              <div
                className="jsx-3877038798 calendar-cell "
              >
                <div
                  className="jsx-3877038798 day"
                >
                  9
                </div>
                <div
                  className="jsx-3877038798 plus"
                  data-testid="add-event-2019-01-10T00:00:00.000Z"
                  onClick={[Function]}
                  role="button"
                >
                  +
                </div>
              </div>
            </td>
            <td
              className="jsx-3390703719"
              data-testid="table-cell-1-3"
            >
              <div
                className="jsx-3877038798 calendar-cell "
              >
                <div
                  className="jsx-3877038798 day"
                >
                  10
                </div>
                <div
                  className="jsx-3877038798 plus"
                  data-testid="add-event-2019-01-11T00:00:00.000Z"
                  onClick={[Function]}
                  role="button"
                >
                  +
                </div>
              </div>
            </td>
            <td
              className="jsx-3390703719"
              data-testid="table-cell-1-4"
            >
              <div
                className="jsx-3877038798 calendar-cell "
              >
                <div
                  className="jsx-3877038798 day"
                >
                  11
                </div>
                <div
                  className="jsx-3877038798 plus"
                  data-testid="add-event-2019-01-12T00:00:00.000Z"
                  onClick={[Function]}
                  role="button"
                >
                  +
                </div>
              </div>
            </td>
            <td
              className="jsx-3390703719"
              data-testid="table-cell-1-5"
            >
              <div
                className="jsx-3877038798 calendar-cell "
              >
                <div
                  className="jsx-3877038798 day"
                >
                  12
                </div>
                <div
                  className="jsx-3877038798 plus"
                  data-testid="add-event-2019-01-13T00:00:00.000Z"
                  onClick={[Function]}
                  role="button"
                >
                  +
                </div>
              </div>
            </td>
            <td
              className="jsx-3390703719"
              data-testid="table-cell-1-6"
            >
              <div
                className="jsx-3877038798 calendar-cell "
              >
                <div
                  className="jsx-3877038798 day"
                >
                  13
                </div>
                <div
                  className="jsx-3877038798 plus"
                  data-testid="add-event-2019-01-14T00:00:00.000Z"
                  onClick={[Function]}
                  role="button"
                >
                  +
                </div>
              </div>
            </td>
          </tr>
          <tr
            className="jsx-3390703719"
          >
            <td
              className="jsx-3390703719"
              data-testid="table-cell-2-0"
            >
              <div
                className="jsx-3877038798 calendar-cell "
              >
                <div
                  className="jsx-3877038798 day"
                >
                  14
                </div>
                <div
                  className="jsx-3877038798 plus"
                  data-testid="add-event-2019-01-15T00:00:00.000Z"
                  onClick={[Function]}
                  role="button"
                >
                  +
                </div>
              </div>
            </td>
            <td
              className="jsx-3390703719"
              data-testid="table-cell-2-1"
            >
              <div
                className="jsx-3877038798 calendar-cell "
              >
                <div
                  className="jsx-3877038798 day"
                >
                  15
                </div>
                <div
                  className="jsx-3877038798 plus"
                  data-testid="add-event-2019-01-16T00:00:00.000Z"
                  onClick={[Function]}
                  role="button"
                >
                  +
                </div>
              </div>
            </td>
            <td
              className="jsx-3390703719"
              data-testid="table-cell-2-2"
            >
              <div
                className="jsx-3877038798 calendar-cell "
              >
                <div
                  className="jsx-3877038798 day"
                >
                  16
                </div>
                <div
                  className="jsx-3877038798 plus"
                  data-testid="add-event-2019-01-17T00:00:00.000Z"
                  onClick={[Function]}
                  role="button"
                >
                  +
                </div>
              </div>
            </td>
            <td
              className="jsx-3390703719"
              data-testid="table-cell-2-3"
            >
              <div
                className="jsx-3877038798 calendar-cell "
              >
                <div
                  className="jsx-3877038798 day"
                >
                  17
                </div>
                <div
                  className="jsx-3877038798 plus"
                  data-testid="add-event-2019-01-18T00:00:00.000Z"
                  onClick={[Function]}
                  role="button"
                >
                  +
                </div>
              </div>
            </td>
            <td
              className="jsx-3390703719"
              data-testid="table-cell-2-4"
            >
              <div
                className="jsx-3877038798 calendar-cell "
              >
                <div
                  className="jsx-3877038798 day"
                >
                  18
                </div>
                <div
                  className="jsx-3877038798 plus"
                  data-testid="add-event-2019-01-19T00:00:00.000Z"
                  onClick={[Function]}
                  role="button"
                >
                  +
                </div>
              </div>
            </td>
            <td
              className="jsx-3390703719"
              data-testid="table-cell-2-5"
            >
              <div
                className="jsx-3877038798 calendar-cell "
              >
                <div
                  className="jsx-3877038798 day"
                >
                  19
                </div>
                <div
                  className="jsx-3877038798 plus"
                  data-testid="add-event-2019-01-20T00:00:00.000Z"
                  onClick={[Function]}
                  role="button"
                >
                  +
                </div>
              </div>
            </td>
            <td
              className="jsx-3390703719"
              data-testid="table-cell-2-6"
            >
              <div
                className="jsx-3877038798 calendar-cell "
              >
                <div
                  className="jsx-3877038798 day"
                >
                  20
                </div>
                <div
                  className="jsx-3877038798 plus"
                  data-testid="add-event-2019-01-21T00:00:00.000Z"
                  onClick={[Function]}
                  role="button"
                >
                  +
                </div>
              </div>
            </td>
          </tr>
          <tr
            className="jsx-3390703719"
          >
            <td
              className="jsx-3390703719"
              data-testid="table-cell-3-0"
            >
              <div
                className="jsx-3877038798 calendar-cell "
              >
                <div
                  className="jsx-3877038798 day"
                >
                  21
                </div>
                <div
                  className="jsx-3877038798 plus"
                  data-testid="add-event-2019-01-22T00:00:00.000Z"
                  onClick={[Function]}
                  role="button"
                >
                  +
                </div>
              </div>
            </td>
            <td
              className="jsx-3390703719"
              data-testid="table-cell-3-1"
            >
              <div
                className="jsx-3877038798 calendar-cell "
              >
                <div
                  className="jsx-3877038798 day"
                >
                  22
                </div>
                <div
                  className="jsx-3877038798 plus"
                  data-testid="add-event-2019-01-23T00:00:00.000Z"
                  onClick={[Function]}
                  role="button"
                >
                  +
                </div>
              </div>
            </td>
            <td
              className="jsx-3390703719"
              data-testid="table-cell-3-2"
            >
              <div
                className="jsx-3877038798 calendar-cell "
              >
                <div
                  className="jsx-3877038798 day"
                >
                  23
                </div>
                <div
                  className="jsx-3877038798 plus"
                  data-testid="add-event-2019-01-24T00:00:00.000Z"
                  onClick={[Function]}
                  role="button"
                >
                  +
                </div>
              </div>
            </td>
            <td
              className="jsx-3390703719"
              data-testid="table-cell-3-3"
            >
              <div
                className="jsx-3877038798 calendar-cell "
              >
                <div
                  className="jsx-3877038798 day"
                >
                  24
                </div>
                <div
                  className="jsx-3877038798 plus"
                  data-testid="add-event-2019-01-25T00:00:00.000Z"
                  onClick={[Function]}
                  role="button"
                >
                  +
                </div>
              </div>
            </td>
            <td
              className="jsx-3390703719"
              data-testid="table-cell-3-4"
            >
              <div
                className="jsx-3877038798 calendar-cell "
              >
                <div
                  className="jsx-3877038798 day"
                >
                  25
                </div>
                <div
                  className="jsx-3877038798 plus"
                  data-testid="add-event-2019-01-26T00:00:00.000Z"
                  onClick={[Function]}
                  role="button"
                >
                  +
                </div>
              </div>
            </td>
            <td
              className="jsx-3390703719"
              data-testid="table-cell-3-5"
            >
              <div
                className="jsx-3877038798 calendar-cell "
              >
                <div
                  className="jsx-3877038798 day"
                >
                  26
                </div>
                <div
                  className="jsx-3877038798 plus"
                  data-testid="add-event-2019-01-27T00:00:00.000Z"
                  onClick={[Function]}
                  role="button"
                >
                  +
                </div>
              </div>
            </td>
            <td
              className="jsx-3390703719"
              data-testid="table-cell-3-6"
            >
              <div
                className="jsx-3877038798 calendar-cell "
              >
                <div
                  className="jsx-3877038798 day"
                >
                  27
                </div>
                <div
                  className="jsx-3877038798 plus"
                  data-testid="add-event-2019-01-28T00:00:00.000Z"
                  onClick={[Function]}
                  role="button"
                >
                  +
                </div>
              </div>
            </td>
          </tr>
          <tr
            className="jsx-3390703719"
          >
            <td
              className="jsx-3390703719"
              data-testid="table-cell-4-0"
            >
              <div
                className="jsx-3877038798 calendar-cell "
              >
                <div
                  className="jsx-3877038798 day"
                >
                  28
                </div>
                <div
                  className="jsx-3877038798 plus"
                  data-testid="add-event-2019-01-29T00:00:00.000Z"
                  onClick={[Function]}
                  role="button"
                >
                  +
                </div>
              </div>
            </td>
            <td
              className="jsx-3390703719"
              data-testid="table-cell-4-1"
            >
              <div
                className="jsx-3877038798 calendar-cell "
              >
                <div
                  className="jsx-3877038798 day"
                >
                  29
                </div>
                <div
                  className="jsx-3877038798 plus"
                  data-testid="add-event-2019-01-30T00:00:00.000Z"
                  onClick={[Function]}
                  role="button"
                >
                  +
                </div>
              </div>
            </td>
            <td
              className="jsx-3390703719"
              data-testid="table-cell-4-2"
            >
              <div
                className="jsx-3877038798 calendar-cell "
              >
                <div
                  className="jsx-3877038798 day"
                >
                  30
                </div>
                <div
                  className="jsx-3877038798 plus"
                  data-testid="add-event-2019-01-31T00:00:00.000Z"
                  onClick={[Function]}
                  role="button"
                >
                  +
                </div>
              </div>
            </td>
            <td
              className="jsx-3390703719"
              data-testid="table-cell-4-3"
            >
              <div
                className="jsx-3877038798 calendar-cell "
              >
                <div
                  className="jsx-3877038798 day"
                >
                  31
                </div>
                <div
                  className="jsx-3877038798 plus"
                  data-testid="add-event-2019-02-01T00:00:00.000Z"
                  onClick={[Function]}
                  role="button"
                >
                  +
                </div>
              </div>
            </td>
            <td
              className="jsx-3390703719"
              data-testid="table-cell-4-4"
            >
              
            </td>
            <td
              className="jsx-3390703719"
              data-testid="table-cell-4-5"
            >
              
            </td>
            <td
              className="jsx-3390703719"
              data-testid="table-cell-4-6"
            >
              
            </td>
          </tr>
        </tbody>
        <div>
          <div
            id="content"
          >
            <div
              className="jsx-4229188709 popup "
              data-testid="popup"
            >
              <div
                className="jsx-4229188709 close"
                data-testid="popup-close"
                onClick={[Function]}
                role="button"
                tabIndex="0"
              >
                x
              </div>
              <div
                className="jsx-4229188709 content"
              >
                <div
                  className="jsx-3897813201 calendar-event"
                  data-testid="form"
                >
                  <input
                    className="jsx-3897813201"
                    data-testid="form-input"
                    type="text"
                  />
                  <textarea
                    className="jsx-3897813201"
                    data-testid="form-textarea"
                  />
                  <button
                    className="jsx-3897813201"
                    data-testid="form-button"
                    onClick={[Function]}
                    type="button"
                  >
                    Create
                  </button>
                </div>
              </div>
            </div>
          </div>
          <div
            data-target-tag-name="BODY"
            id="target"
          />
        </div>
      </table>
    </div>
  </div>
</div>
`;
